web/packages/teleterm/src/mainProcess/rootClusterProxyHostAllowList.ts

```diff
@@ -77,22 +77,23 @@ export function manageRootClusterProxyHostAllowList({
 
     allowList.clear();
     for (const rootCluster of rootClusters) {
-      if (!rootCluster.proxyHost) {
-        continue;
+      if (rootCluster.proxyHost) {
+        let browserProxyHost: string;
+        try {
+          browserProxyHost = proxyHostToBrowserProxyHost(rootCluster.proxyHost);
+          allowList.add(browserProxyHost);
+        } catch (error) {
+          logger.error(
+            'Ran into an error when converting proxy host to browser proxy host',
+            error
+          );
+        }
       }
 
-      let browserProxyHost: string;
-      try {
-        browserProxyHost = proxyHostToBrowserProxyHost(rootCluster.proxyHost);
-      } catch (error) {
-        logger.error(
-          'Ran into an error when converting proxy host to browser proxy host',
-          error
-        );
-        continue;
+      // Allow the SSO hostname for SSO login/mfa redirects.
+      if (rootCluster.ssoHostname) {
+        allowList.add(rootCluster.ssoHostname);
```

Comment:
When this function executes the following above:

```typescript
browserProxyHost = proxyHostToBrowserProxyHost(rootCluster.proxyHost);
allowList.add(browserProxyHost);
```

It doesn't add a proxy hostname to the allow list – it adds a host which includes the port number.

Similarly, this shouldn't be a hostname, it should be a host. The code doesn't work for my local cluster, because the MFA prompt tries to open a host which is not in the allowlist. The prompt tries to open `enterprise-leaf:3040`, while the allowlist only has `enterprise-leaf`.

Reply:
Nice catch, fixed.